src/ansible_builder/_target_scripts/introspect.py

```diff
@@ -203,11 +216,24 @@ def simple_combine(reqs):
                 continue
 
             base_line = line.split('#')[0].strip()
+            name_match = REQ_NAME_RE.match(base_line)
```

Comment:
Some additional care is probably required here.  This will have issues currently with things like:

* URLs (`git+...`, `https://...`)
* `-r extra-requirements.txt` or any other random pip flag

In this case, we probably just need to skip the exclude checks. But we can't have it both ways, in that we can't ignore requirements parsing, and still do requirements parsing, so URLs and such just wouldn't be supported. 🙂 

The regex is also using the pep503 info, but I can't think of reasons it wouldn't work with RPM package names.